src/v1/spatial-types.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {int} from './integer';
+
+const POINT_IDENTIFIER_PROPERTY = '__isPoint__';
+
+/**
+ * Represents a single two or three-dimensional point in a particular coordinate reference system.
+ */
+export class Point {
+
+  /**
+   * @constructor
+   * @param {number|Integer} srid the coordinate reference system identifier.
+   * @param {number} x the <code>x</code> coordinate of the point.
+   * @param {number} y the <code>y</code> coordinate of the point.
+   * @param {number|undefined} z the <code>y</code> coordinate of the point or <code>undefined</code> if point has 2 dimensions.
+   */
+  constructor(srid, x, y, z) {
```

Comment:
If we have `p1 = new Point(1, 1, 0)`, a 2D point after construction and later on do something like `p1.z = 4` we'll end up with a 3D point. I think it would be better if we could keep point's dimensions immutable, basically deciding it in constructor and not exposing it as a public property. Does it make sense?